examples/custom/src/reserver.rs

```diff
@@ -0,0 +1,41 @@
+// This file is part of Gear.
+
+// Copyright (C) 2023-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(not(feature = "std"))]
+pub(crate) mod wasm {
+    use gstd::{msg, prelude::*, ReservationId};
+
+    #[derive(Default)]
+    pub(crate) struct State {
+        reservation_id: Option<ReservationId>,
+    }
+
+    pub(crate) fn init() -> State {
+        Default::default()
+    }
```

Comment:
useless

Reply:
It is not the `init` entrypoint. This function returns state to be used here: 

https://github.com/gear-tech/gear/pull/3698/files#diff-7338aaa9606870091a816d7873106f089b9cc8465c00bd8dfc476ced27002d55R87

Reply:
But we can replace it by creating `State` in place indeed.

Reply:
[af7e257](https://github.com/gear-tech/gear/pull/3698/commits/af7e25764c7fe81df19b160a3d1665be54f0ac3b)